DIPs/DIP1xxx-rval_to_ref.md

```diff
@@ -0,0 +1,345 @@
+# `ref T` accepts r-values
+
+| Field           | Value                                                           |
+|-----------------|-----------------------------------------------------------------|
+| DIP:            | (number/id -- assigned by DIP Manager)                          |
+| Review Count:   | 0 (edited by DIP Manager)                                       |
+| Author:         | Manu Evans (turkeyman@gmail.com)                                |
+| Status:         | Draft                                                           |
+
+## Abstract
+
+A recurring complaint from users when interacting with functions that receive arguments by `ref` is that given an rvalue as argument, the compiler is unable to create an implicit temporary to perform the function call, and presents the user with a compile error instead.  
+This situation leads to a workaround where function parameters must be manually assigned to temporaries prior to the function call, which many users find frustrating.
+
+A further issue is that because this situation require special-case handling, this may introduce semantic edge-cases and necessitate undesirable compile-time logic invading the users code, particularly into generic code.
+
+`ref` args are not as common in conventional idiomatic D as they are in some other languages, but they exist and appear frequently in niche circumstances. As such, this issue is likely to disproportionately affect subsets of users who find themselves using `ref` arguments more than average.
+
+The choice to receive an argument by value or by reference is a detail that the API *author* selects with respect to criteria relevant to their project or domain, however, the semantic impact is not worn by the API author, but rather by the API user, who may be required to jump through hurdles to interact the API with their local code.  
+It would be ideal if the decision to receive arguments by value or by reference were a detail for the API, and not increase the complexity of the users code.
+
+Here is proposed a strategy to emit implicit temporaries to conveniently interact with APIs that use `ref` arguments.
+
+## Reference
```

Comment:
It's great that you've listed so many references, but it's a bit much for anyone coming to all of this for the first time without anything to guide them. A one-line summary for each reference link (forum threads, issues, and PRs all) would be extremely helpful here.